test/integration/test_messages_api.py

```diff
@@ -0,0 +1,184 @@
+        self.assertIsInstance(api_response.priority, PriorityEnum)
+        self.assertEqual(api_response.priority, self.priority)
+        self.assertEqual(api_response.segment_count, 1)
+        self.assertTrue(datetime.fromisoformat(api_response.time[:-1]))
+
+
+    def test_create_message_bad_request(self):
+        with self.assertRaises(ApiException) as context:
+            self.api_instance.create_message(self.account_id, self.invalid_message_request)
+            
+        self.assertEqual(context.exception.status, 400)
+
+        e = json.loads(context.exception.body)
+        self.assertEqual(e['type'], 'request-validation')
+        self.assertIsInstance(e['description'], str)
+        self.assertIsInstance(e['fieldErrors'], list)
+
+        field_error = e['fieldErrors'][0]
+        self.assertIsInstance(field_error['fieldName'], str)
+        self.assertIsInstance(field_error['description'], str)
+    
+
+    def test_create_message_unauthorized(self):
+        with self.assertRaises(UnauthorizedException) as context:
+            self.unauthorized_api_instance.create_message(self.account_id, self.invalid_message_request)
+             
+        self.assertEqual(context.exception.status, 401)
+
+        e = json.loads(context.exception.body)
+        self.assertEqual(e['type'], 'unauthorized')
+        self.assertEqual(e['description'], 'Authentication Failed')
+
+    
+    @unittest.skip('The SDK catches incorrect content-type before making the request and attempts to create an ApiException,\
+                    but the creation of the exception fails since there is no response body. This should probably create some\
+                    kind of Client Exception instead, since this is not an actual API Exception.')
+    def test_create_message_invalid_media(self):
+        with self.assertRaises(ApiException) as context:
+            self.api_instance.create_message(self.account_id, self.message_request, _content_type='application/xml')
+
+
+    def test_list_messages(self):
+        message_direction = ListMessageDirectionEnum("OUTBOUND")
+        
+        response = self.api_instance.list_messages(self.account_id, message_direction=message_direction, _return_http_data_only=False)
+
+        self.assertEqual(response[1], 200)
+
+        api_response = response[0]
+        self.assertIsInstance(api_response, MessagesList)
+        self.assertGreater(api_response.total_count, 0)
+        self.assertTrue(api_response.messages[0], ListMessageItem)
+        
+        message = api_response.messages[0]
+        self.assertEqual(message.account_id, self.account_id)
+        self.assertRegex(message.destination_tn, '^\\+[1-9]\\d{1,14}$')
+        self.assertIsInstance(message.message_direction, ListMessageDirectionEnum)
+        self.assertTrue(message.message_id)
+        self.assertIsInstance(message.message_status, MessageStatusEnum)
+        self.assertIsInstance(message.message_type, MessageTypeEnum)
```

Comment:
Same as above, check values as it won't be dynamic.

Reply:
This _probably_ won't be dynamic, but we can't guarantee what message comes first in the list, since this is listing all outbound messages on account.